thoth/adviser/resolver.py

```diff
@@ -1328,7 +1333,7 @@ def _do_resolve_states(
             self.context.accepted_final_states_count / duration,
         )
 
-        if max_score is not None:
+        if self.beam.size > 0:
```

Comment:
Could we leave this one as it is, or will it still cause issues with mypy?

Reply:
mypy won't complain (I guess there is no type constraints on `is` condition in Python ?), but that won't work : `max_score is not None` is always True since we now use '-inf' as a starting value.
From my understanding, `self.beam.size` is the number of "rated software stack" right ? That's how I was translating the condition in my head. Maybe adding a comment would explicit it ?

We could have `max_score is not float("-inf")` I guess ?
That's another way of checking for the same thing. My rationale for using `self.beam.size` is in the last commit message: 
`max_score is float("-inf")` is a consequence of `self.beam.size == 0`, not the other way. aka, we don't have a score for the top rated software stack because we have no rated software stack.

But I don't care much either way, I'll check for -inf if you prefer it.

Reply:
So going back to the code, I think `max_score != float("-inf")` is in fact the right option. Reading the generators involved, I can't confidently find the relation between the two. -> better be safe.